ls_handleTouches.ino

```diff
@@ -1732,18 +1732,23 @@ void handleTouchRelease() {
       preSendLoudness(sensorSplit, 0, 0, sensorCell->note, sensorCell->channel);
     }
 
-    // unregister the note <> cell mapping
-    if (!isSwitchLegatoPressed(sensorSplit) && (!isArpeggiatorEnabled(sensorSplit) || !isSwitchLatchPressed(sensorSplit))) {
-      noteTouchMapping[sensorSplit].noteOff(sensorCell->note, sensorCell->channel);
-    }
-
+    // See if we should unregister the note <> cell mapping
+    boolean shouldUnregisterNote = !isSwitchLegatoPressed(sensorSplit) && (!isArpeggiatorEnabled(sensorSplit) || !isSwitchLatchPressed(sensorSplit));
+    
     // send the Note Off
     if (isArpeggiatorEnabled(sensorSplit)) {
       if (!isSwitchLatchPressed(sensorSplit)) {
         handleArpeggiatorNoteOff(sensorSplit, sensorCell->note, sensorCell->channel);
       }
+      else if(shouldUnregisterNote) {
```

Comment:
This is what I didn't like the look of and was actually unsure if it was necessary to keep. If we unfold the logic from above, this doesn't seem like a possible path.